src/Symfony/Bundle/FrameworkBundle/Command/SecretsGenerateKeyCommand.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Exception\EncryptionKeyNotFoundException;
+use Symfony\Bundle\FrameworkBundle\Secret\EncoderInterface;
+use Symfony\Bundle\FrameworkBundle\Secret\MutableSecretStorageInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @author Tobias Schultze <http://tobion.de>
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+final class SecretsGenerateKeyCommand extends Command
+{
+    protected static $defaultName = 'secrets:generate-key';
+    private $secretsStorage;
+    private $encoder;
+
+    public function __construct(EncoderInterface $encoder, MutableSecretStorageInterface $secretsStorage)
+    {
+        $this->secretsStorage = $secretsStorage;
+        $this->encoder = $encoder;
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setDefinition([
+                new InputOption('force', 'f', InputOption::VALUE_NONE, 'Override previous key.'),
```

Comment:
IMO there should be two separate options:
- `overwrite` which overwrites the key without re-encrypting the existing secrets
- `rekey` which generates new keys and re-encrypts the existing secrets with the new key (rekey is the term used for this, see https://en.wikipedia.org/wiki/Rekeying_(cryptography) and https://www.vaultproject.io/api/system/rekey.html

Instead of having a --rekey option, it might make sense to have this as a separate command instead.

Reply:
I renamed the option in `rekey`

I'm not sure for ce `overwrite` option. Do you have a use case to rotate keys without re-encrypting the previous content? Such option would let the store in a inconsistent state with a mix of secret encoded with 2 different keys and not decryptable.